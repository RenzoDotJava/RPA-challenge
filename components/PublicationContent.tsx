import { View, Text, Platform, StyleSheet } from "react-native";

interface PublicationContentProps {
	index: number;
	title: string;
	body: string;
}

const PublicationContent = ({
	index,
	title,
	body,
}: PublicationContentProps) => {
	return (
		<View style={styles.container}>
			<View
				style={[
					styles.titleWrapper,
					{ borderTopWidth: index == 0 ? 0 : 2 },
				]}
			>
				<Text style={styles.title}>{title}</Text>
			</View>
			<Text style={styles.body}>{body}</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		backgroundColor: "#DFDFDF",
		paddingHorizontal: 13,
	},
	titleWrapper: {
		paddingTop: 15,
		borderTopColor: "#C9C9C9",
	},
	title: {
		color: "#32373A",
		fontWeight: "bold",
		fontSize: 16,
		fontFamily: Platform.OS === "android" ? "serif" : "Arial",
	},
	body: {
		fontSize: 14,
		paddingRight: 30,
		color: "#737475",
		fontWeight: "500",
		marginTop: 8,
		paddingBottom: 15,
	},
});

export default PublicationContent;
