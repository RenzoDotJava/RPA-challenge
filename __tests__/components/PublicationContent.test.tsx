import { render, screen } from "@testing-library/react-native";
import PublicationContent from "../../components/PublicationContent";

const props = {
	index: 1,
  title: 'title',
  body: 'body'
};

describe("Testing <PublicationTitle />", () => {
	it("Matching snapshot", () => {
		render(<PublicationContent {...props} />);

		expect(screen).toMatchSnapshot();
	});

	it("Showing the props", () => {
		render(<PublicationContent {...props} />);

		expect(screen.getByText(props.title)).toBeTruthy();
		expect(screen.getByText(props.body)).toBeTruthy();
	});
});