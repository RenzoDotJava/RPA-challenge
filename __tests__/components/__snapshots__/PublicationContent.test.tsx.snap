// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <PublicationTitle /> Matching snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#DFDFDF",
      "paddingHorizontal": 13,
    }
  }
  testID="publication-content"
>
  <View
    style={
      [
        {
          "borderTopColor": "#C9C9C9",
          "paddingTop": 15,
        },
        {
          "borderTopWidth": 2,
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#32373A",
          "fontFamily": "Arial",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      title
    </Text>
  </View>
  <Text
    style={
      {
        "color": "#737475",
        "fontFamily": "Arial",
        "fontSize": 14,
        "fontWeight": "500",
        "marginTop": 8,
        "paddingBottom": 15,
        "paddingRight": 30,
      }
    }
  >
    body
  </Text>
</View>
`;
